boringtun/src/device/mod.rs

```diff
@@ -1068,35 +1073,34 @@ impl Device {
                         None => continue,
                     };
 
-                    let mut peer = match peers.find(dst_addr) {
-                        Some(peer) => peer.lock(),
+                    let peer = match peers.find(dst_addr) {
+                        Some(peer) => peer,
                         None => continue,
                     };
 
+
                     if let Some(callback) = &d.config.firewall_process_outbound_callback {
-                        if !callback(&peer.tunnel.peer_static_public().to_bytes(), src) {
+                        if !callback(&peer.public_key.0, src) {
                             continue;
                         }
                     }
 
-                    let mut public_key = String::with_capacity(32);
-                    for byte in peer.tunnel.peer_static_public().as_bytes() {
-                        let pub_symbol = format!("{:02X}", byte);
-                        public_key.push_str(&pub_symbol);
-                    }
-
-                    match peer.tunnel.encapsulate(src, &mut t.dst_buf[..]) {
+                    let res = {
+                        let mut tun = peer.tunnel.lock();
+                        tun.encapsulate(src, &mut t.dst_buf[..])
+                    };
```

Comment:
Same, `let res = peer.tunnel.lock().encapsulate(..);` is ok here. Just don't inline `res` into a `match`.